keps/sig-storage/1495-volume-populators/README.md

```diff
@@ -271,6 +294,153 @@ downgrade.
 
 No issues
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: AnyVolumeDataSource
+    - Components depending on the feature gate: kube-apiserver
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+  Before this feature, kube API server would silently drop any PVC data source
+  that it didn't recognize, as if the client never populated the field. After
+  enabling this feature, the API server allows any object to be specified.
+  An external admission controller will take over validation of the field,
+  and its response to invalid data sources will be to fail the create operation,
+  rather than silently drop them.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Turning off the feature gate should drop all the data sources that were not
+  already valid and restore the old behavior.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  PVC which previously had data sources that were invalided by the rollback
+  would be empty volumes. Users would need to delete and re-create those PVCs.
```

Comment:
Is this specific to re-enablement or is this the impact of disabling?

Reply:
Yes, this is a weird side effect of the existing admission controller which silently drops data sources it doesn't like rather than failing the requests.

Reply:
Have you tested to see if you can delete them?  The deletion process requires updates, which mean validation has to pass.

Can you describe what happens to the pods using the PVCs which previously existed?